### PR TITLE
Maint: update serverapi-openapi dependency

### LINKED ISSUE #133: Update serverapi-openapi dependency to the 2024 R1 release version
Once https://github.com/ansys/grantami-serverapi-openapi/issues/94 is complete, we should release a new version of this package that uses it.

There are no functional changes expected to the underlying API required by this package, and the CI is currently working against a 2024 R1 development build, but we should still release a new version.

### LINKED ISSUE #139: Update supported Python versions
Drop Python 3.8, support Python 3.12

### PR DESCRIPTION
**Pull request by ludovicsteinbach**
Closes #133
Closes #139

Update the serverapi-openapi dependency:
- No API definition changes for record lists (24.1.455)
- Update method calls to use keyword only args
- Update Enum definition and associated tests

serverapi-openapi 2.0 dropped Python 3.8 so this PR also drops support for Python 3.8 and adds Python 3.12

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -50,7 +50,7 @@ Dependencies
 
 To use the ``ansys.grantami.recordlists`` package you must have access to a Granta MI 2023 R2 deployment.
 
-The ``ansys.grantami.recordlists`` package currently supports Python from version 3.8 to version 3.11.
+The ``ansys.grantami.recordlists`` package currently supports Python from version 3.9 to version 3.11.
```
Comment (Andy-Grigg) [inline README.rst:53]:
```suggestion
The ``ansys.grantami.recordlists`` package currently supports Python from version 3.9 to version 3.12.
```

File: tests/test_models.py [hunk 2]
```diff
@@ -319,7 +321,7 @@ def test_nested_boolean_criterion_dto_mapping(self):
         leaf_criterion = nested_boolean_criterion.match_all[0]
         assert isinstance(leaf_criterion, GrantaServerApiListsDtoRecordListSearchCriterion)
         assert leaf_criterion.name_contains == "A"
-        assert leaf_criterion.user_role == GrantaServerApiListsDtoUserRole.OWNER
+        assert leaf_criterion.user_role.value == GrantaServerApiListsDtoUserRole.OWNER.value
```
Comment (Andy-Grigg) [inline tests/test_models.py:324]: Would

```suggestion
        assert leaf_criterion.user_role == GrantaServerApiListsDtoUserRole.OWNER
```

Not also work?

**Review by Andy-Grigg — CHANGES_REQUESTED** [review]
Couple of very minor changes.

File: tests/test_models.py (hunk 2 above)
Comment (ludovicsteinbach) [reply tests/test_models.py:324]: Interesting question, I wasn't sure, but I've checked and no, it's not equivalent.
Members of different enums are not equal even if they have the same value.
The assertion could be:
```assert leaf_criterion.user_role == UserRole.OWNER```

I guess that's what you meant actually?

Comment (Andy-Grigg) [reply tests/test_models.py:324]: Yes, I think so. Comparing the enum member itself, as opposed to the value of the member.

Comment (ludovicsteinbach) [reply tests/test_models.py:324]: I'm sorry you were right, I had misunderstood the test. We should expect the result to be `GrantaServerApiListsDtoUserRole.OWNER`. It is not and that's a bug. I've not updated the _to_model method to correctly instantiate an enum.

**Review by Andy-Grigg — APPROVED** [review]

**Pull request merged**